tests/datastore/test_sources.py

```diff
@@ -107,14 +108,38 @@ def test_timestamp_format_inference(rundb_mock):
 
 
 @pytest.mark.parametrize(
-    "additional_filters",
-    [[("age", "=", float("nan"))], [("age", "in", [10, float("nan")])]],
+    "additional_filters, message",
+    [
+        ([("x", "=", 3)], ""),
+        (
+            [[("x", "=", 3), ("x", "=", 4), ("x", "=", 5)]],
+            "mlrun supports additional_filters only as a list of tuples.",
+        ),
+    ],
 )
-def test_nan_additional_filters(additional_filters):
-    with pytest.raises(
-        mlrun.errors.MLRunInvalidArgumentError,
-        match="using NaN in additional_filters is not supported",
-    ):
+def test_transform_list_filters_to_tuple(additional_filters, message):
+    after_json_change_filters = json.loads(json.dumps(additional_filters))
```

Comment:
Same suggestion again
```suggestion
    back_from_json_serialization = json.loads(json.dumps(additional_filters))
```